packages/notifications/push/amplify_push_notifications/android/src/main/kotlin/com/amazonaws/amplify/amplify_push_notifications/PushNotificationFirebaseMessagingService.kt

```diff
@@ -67,15 +76,17 @@ class PushNotificationFirebaseMessagingService : FirebaseMessagingService() {
                     utils.showNotification(
                         payload, baseContext.getLaunchActivityClass()
                     )
-
                     Log.i(
                         TAG, "App is in background, start background service and enqueue work"
                     )
-                    FlutterMain.startInitialization(baseContext)
-                    FlutterMain.ensureInitializationComplete(baseContext, null)
-                    PushNotificationBackgroundService.enqueueWork(
-                        baseContext, remoteMessage.toIntent()
-                    )
+                    val mainEngine = FlutterEngineCache.getInstance()
+                        .get(PushNotificationPluginConstants.FLUTTER_ENGINE_ID)
+                    if (mainEngine == null) {
+                        engineGroup.createAndRunDefaultEngine(baseContext)
```

Comment:
I would add comments as doc here to describe that this runs from `lib/main.dart` if the App is at the killed state at this moment and why, or wrap this in a method `runAppFromKilledState` or something similar to make it explicit.